File: Algorithm/binary/binarySearch.py
import random
data_list = random.sample(range(50), 10)


def binary_search(data, search):
  data.sort()
  print(data)

  if len(data) == 1 and search == data[0]:
    return True
  
  if len(data) == 1 and search != data[0]:
    return False
  
  if len(data) == 0:
    return False
  
  mid = len(data) // 2
  if search == data[mid]:
    return True
  else:
    if search > data[mid]:
      return binary_search(data[mid:], search)
    else:
      return binary_search(data[:mid], search)


print(binary_search(data_list, 18))
"""
[0, 2, 5, 8, 19, 22, 23, 27, 32, 34]
[0, 2, 5, 8, 19]
[5, 8, 19]
[8, 19]
[8]
False
"""

File: Algorithm/binary/binarySearch.js
// 이진탐색

data_list = [11, 2, 54, 23, 33, 12, 6];

const sorted = (arr) => {
  return arr.sort((a, b) => a - b); // 오름차순정렬
};

const binarySearch = function (arr, search) {
  
  while(mid == search)
    let mid = Math.floor(arr.length / 2);

    if (!arr.length) return false; // 배열의 길이가 0이면

    if (arr[mid] == search) {
      return true;
    }
    if (arr[mid] > search) {
      return binarySearch(arr.slice(0, mid), search);
    } else return binarySearch(arr.slice(mid + 1), search);
};

console.log(binarySearch(sorted(data_list), 23)); // true
console.log(binarySearch(sorted(data_list), 87)); // false
